src/common/hooks/useDialog/index.tsx

```diff
@@ -0,0 +1,38 @@
+import dynamic from 'next/dynamic'
+
+import { useState } from 'react'
+
+import { UniversalDialogProps } from '@/common/components/UniversalDialog/types'
+
+import { UseDialog } from './types'
+
+const DynamicUniversalDialog = dynamic(
+  async () =>
+    (
+      await import(
+        /* webpackChunkName: "UniversalDialog" */
+        '@/common/components/UniversalDialog'
+      )
+    ).UniversalDialog,
+  {
+    ssr: false,
+  }
+)
+
+export const useDialog = (initialProps?: Omit<UniversalDialogProps, 'open' | 'setOpen'>): UseDialog => {
```

Comment:
I don't agree with creating this hook. It's way cleaner to just use `useState` and render the `UniversalDialog` tag directly.